rust/libnewsboat/src/utils.rs

```diff
@@ -477,6 +479,19 @@ pub fn getcwd() -> Result<PathBuf, io::Error> {
     env::current_dir()
 }
 
+/// Recursively create directories if missing and set permissions accordingly.
+pub fn mkdir_parents(p: &str, mode: u32) -> isize {
+    let path = Path::new(p);
+    let result = fs::DirBuilder::new()
+        .mode(mode)
+        .recursive(true) // directories created with same security and permissions
+        .create(path);
+    match result {
```

Comment:
I think it would be better to do this conversion in `libnewsboat-ffi`. This way, our C++ code can keep working with a (somewhat uninformative) integer, and our Rust code is forced to handle the `Result` properly.

Reply:
Totally! I'm just creating more unnecessary work haha. I've moved the conversions over.